### PR TITLE
Implement struct valued constant validator for loading schemas

### LINKED ISSUE #466: Struct-valued constants are not yet implemented
As it stands today, the current thrifty schema is not able to parse struct valued constants. Wanted to know if this is a work in progress? If not, was this something consciously left out from implementation?

I am trying to load schemas for manipulating certain exceptions based on added annotations. Given that we only care about exceptions structs, is it possible to silently fail the constant schema generation so that it does not throw `IllegalStateException`? 

Several of our thrift definitions have struct valued constants so wondering if there is any workaround?

### PR DESCRIPTION
**Pull request by janarajan**
This commit resolves #466

### CODE REVIEW

File: thrifty-schema/src/test/kotlin/com/microsoft/thrifty/schema/LoaderTest.kt [hunk 1]
```diff
@@ -992,6 +995,172 @@ class LoaderTest {
...
+        val thrift = """
+            
+            struct Region {
+                1: required string name 
+                2: optional bool isActive
+            }
+
+            const Region DEFAULT_REGION = {
+                1 : "US",
+                "isActive" : true // field name does not exist in Region
```
Comment (benjamin-bader) [inline thrifty-schema/src/test/kotlin/com/microsoft/thrifty/schema/LoaderTest.kt:1104]: `isActive` does in fact exist; should the comment be for the line above this one?

**Review by benjamin-bader — CHANGES_REQUESTED** [review]
Nice work!  A few small nits but this is basically good-to-go.

File: thrifty-schema/src/main/kotlin/com/microsoft/thrifty/schema/Constant.kt [hunk 2]
```diff
@@ -142,6 +143,11 @@ class Constant private constructor (
                 return MAP
             }
 
+            if (tt.isStruct) {
+                // this should work for exception type as well. structType has isException field
+                return STRUCT
+            }
+
             throw IllegalStateException("Struct-valued constants are not yet implemented")
```
[the remark this answers is not in the document]
Comment (janarajan) [reply thrifty-schema/src/main/kotlin/com/microsoft/thrifty/schema/Constant.kt]: :) Yup, fixed the messaging

**Comment by benjamin-bader** [comment]
Thanks!  Looks good; assuming tests pass, please [sign the CLA](https://cla.opensource.microsoft.com/microsoft/thrifty?pullRequest=467) and we'll be good to merge.

**Review by ravirajj — APPROVED** [review]

File: thrifty-schema/src/main/kotlin/com/microsoft/thrifty/schema/Constant.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ravirajj) [reply thrifty-schema/src/main/kotlin/com/microsoft/thrifty/schema/Constant.kt]: nevermind... the message was updated in the latest update!

**Review by benjamin-bader — APPROVED** [review]

**Comment by janarajan** [comment]
@benjamin-bader thanks for the review and approval. I have now signed the CLA so you can accept and merge the PR. Let me know.

**Comment by janarajan** [comment]
@benjamin-bader thanks for merging the PR. What is the release cycle? Any possibility of releasing a candidate with this change?

**Comment by benjamin-bader** [comment]
There's no official schedule; releases happen whenever there's enough to justify a release.  I'll see about making an RC soon but am not confident I have my laptop properly configured for it right now - it might take a little while.

In the meantime I believe you can use a snapshot build.

**Comment by benjamin-bader** [comment]
Version 3.1.0-RC01 should be up in Maven Central in the next 10 minutes or so

**Pull request merged**